drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -147,10 +147,49 @@ static int lis2dh_acc_odr_set(const struct device *dev, uint16_t freq)
 	odr = lis2dh_freq_to_odr_val(freq);
 	if (odr < 0) {
 		return odr;
+	}else if (odr == 0 && data->powered_down == false){
+		/* check if high resolution mode is enabled as it is not allowed to
```

Comment:
The primary way to select the operation mode (low-power / normal / high res) is done through the kconfig. This PR's added functionality is a way for the application to make the sensor go into a full power down mode. 
When the driver sets the ODR to 0 (which the datasheet describes as 'Power down mode') it tries to save as much power as possible by also putting the sensor in low power mode (regardless of what is selected in the kconfig) and disabling all axis as well. 
When the ODR is then set to a value larger than 0 the driver restores the previous (kconfig selected) operating mode and re-enables the axis. 

This functionality could be put in a dedicated function but this integration felt more streamlined to me. Besides that the datasheet's description of 'power down mode' is tied with the ODR, so retaining that connection in the code felt fitting.

Reply:
As we discussed in the meeting on Tuesday, I'm still not sure if doing it like this is the best way. The drivers have device power management capabilities, and if you still have time it would be very beneficial to look into this. Also including stuff like turning off axes, which can greatly impact the power consumption (but this could potentially fit better in the init function).
I understand that the ODR and power down mode are interlinked, so if possible please think about the best solution keeping also the above things in mind

Reply:
The power down functionality is now moved to a dedicated function. The option to enable/disable individual axis is on my todo list.